meson.build

```diff
@@ -1,34 +1,51 @@
 project('ashuffle', ['c', 'cpp'], default_options: ['cpp_std=c++17', 'warning_level=2'])
 
 cmake = import('cmake')
+cpp = meson.get_compiler('cpp')
 
-absl = cmake.subproject('absl', cmake_options: [
-  '-DCMAKE_CXX_STANDARD=17',
-])
-
-# Note: Unfortunately, meson does not pull in the "dependent" libraries
-# correctly so we have to list them here. If you take a dependency on some
-# addtional absl features, you will probably need to add more `dependency`
-# lines here. This could also be broken on upgrades.
-absl_all = [
+# absl dependencies need to be explicited...
+# It might be possible to use cmake dependencies (e.g. "absl:string"
+# defined in abslTargets.cmake in the future but that does not seem
+# worth the time trying to figure that out.
+absl_libs = [
     # Via Base:
-    absl.dependency('raw_logging_internal'),
+    'raw_logging_internal',
 
     # Via Strings:
-    absl.dependency('int128'),
-    absl.dependency('str_format_internal'),
-    absl.dependency('strings_internal'),
-    absl.dependency('strings'),
+    'int128',
+    'str_format_internal',
+    'strings_internal',
+    'strings'
 ]
 
+absl_deps = []
+if get_option('absl_subproject')
```

Comment:
I would prefer if the option to use the system absl disabled by default, and users had to enable the option. Also a name that made it clearer this was "at your own risk" would be nice. I like `unsupported_use_system_absl` as I mentioned in a comment.

Reply:
Done!